scripts/pre_commit_hook.py

```diff
@@ -74,16 +75,17 @@ def install_hook():
             python_utils.PRINT('Created symlink in .git/hooks directory')
         # Raises AttributeError on windows, OSError added as failsafe.
         except (OSError, AttributeError):
-            shutil.copy(__file__, pre_commit_file)
+            shutil.copy(this_file, pre_commit_file)
             python_utils.PRINT('Copied file to .git/hooks directory')
 
     python_utils.PRINT('Making pre-commit hook file executable ...')
-    _, err_chmod_cmd = start_subprocess_for_result(chmod_cmd)
+    if not common.is_windows_os():
```

Comment:
Just to check -- you've manually tested that the pre-commit hook does work correctly on Windows with this change, right?

Reply:
If the default python is python2, then it works correctly. Otherwise, there will be an error message. But since the most time we are in a virtual environment, it should be fine.